pkg/controller/actions/gc/engine/gc_engine.go

```diff
@@ -136,15 +132,16 @@ func (gc *GC) listResources(
 func (gc *GC) Run(
 	ctx context.Context,
 	cli *odhcli.Client,
-	selector labels.Selector,
 	opts ...RunOptionsFn,
 ) (int, error) {
+	l := logf.FromContext(ctx)
+
 	ro := runOptions{
 		typePredicate: func(_ context.Context, _ schema.GroupVersionKind) (bool, error) {
 			return true, nil
 		},
 		objectPredicate: func(_ context.Context, _ unstructured.Unstructured) (bool, error) {
-			return true, nil
+			return false, nil
```

Comment:
why this default behavior fliped to "false"?

Reply:
The engine now collects only objects matching the object predicate, preventing unintended deletions of **everything** by default.